packages/gestalt/src/List.js

```diff
@@ -0,0 +1,133 @@
+   */
+  labelDisplay?: 'visible' | 'hidden',
+  /**
+   * The spacing used in List. See the [spacing variant](https://gestalt.pinterest.systems/web/list#Spacing) for guidance.
+   */
+  size?: 'regular' | 'condensed',
+  /**
+   * Determines the style of the list. See the [type variant](https://gestalt.pinterest.systems/web/list#Type) to learn more.
+   */
+  type?: ListType,
+|};
+
+/**
+ * [List](https://gestalt.pinterest.systems/web/list) allows users to view individual, but related, text items grouped together.
+ * ![List light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/List.spec.mjs-snapshots/List-chromium-darwin.png)
+ * ![List dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/List-dark.spec.mjs-snapshots/List-dark-chromium-darwin.png)
+ */
+function List({ label, labelDisplay = 'visible', size = 'regular', type, children }: Props): Node {
+  const { type: inheritedType, style: inheritedStyle } = useList();
+  const id = useId();
+
+  let listType: ?ListType = type;
+
+  if (!listType && inheritedType) {
+    listType = inheritedType;
+  }
+
+  if (!listType) {
+    listType = 'unordered';
+  }
+
+  const ListElement = listType === 'ordered' ? 'ol' : 'ul';
+
+  const listChildren = getChildrenToArray({ children, filterLevel: 'List' });
+
+  const didTypeChanged = !!inheritedType && !!type && inheritedType !== type;
+
+  const newInheritedStyleOl =
+    !didTypeChanged && listType === 'ordered' ? inheritedStyle?.ol.slice(1) : inheritedStyle?.ol;
+
+  const newInheritedStyleUl =
+    !didTypeChanged && listType === 'unordered' ? inheritedStyle?.ul.slice(1) : inheritedStyle?.ul;
+
+  return (
+    <ListProvider
+      type={listType}
+      size={size}
+      style={
+        inheritedStyle
+          ? {
+              ol: newInheritedStyleOl ?? [],
+              ul: newInheritedStyleUl ?? [],
+            }
+          : { ol: STYLE_SEQUENCE_ORDERED, ul: STYLE_SEQUENCE_UNORDERED }
+      }
+    >
+      <NestingProvider componentName="List" maxNestedLevels={6}>
+        <Flex direction="column">
+          <Box
+            id={id}
+            display={labelDisplay === 'hidden' ? 'visuallyHidden' : 'block'}
+            marginBottom={labelDisplay === 'hidden' ? 0 : 3}
+          >
+            <ListText text={label} />
+          </Box>
+          <Text>
+            <ListElement
+              aria-labelledby={id}
+              className={classnames(styles.list, {
+                [styles.bareList]: type === 'bare',
+                [styles.regular]: type === 'regular',
+                [styles.condensed]: type === 'condensed',
+              })}
+            >
+              {listChildren}
+            </ListElement>
+          </Text>
+        </Flex>
+      </NestingProvider>
+    </ListProvider>
+  );
+}
+
+List.displayName = 'List';
+
+List.Item = ListItem;
+List.NestedList = NestedList;
```

Comment:
I created NestedList because List has more props, and NestedList just focuses on the type. 

Under the hood List & NestedList are the same but Lists control more styling (label, spacing, etc) while List doesn't

Reply:
There's a lot of repeated logic between the two components. Can they be combined to a single internal component, then exposed with different APIs?